Vhost and upload definition from overview view

tbroden84 opened the pull request:
From detected issue regarding uploading definitions from the UI and monitoring tag was required. 

See https://github.com/cloudamqp/lavinmq/pull/408

Overview view and upload functionality tried to upload definitions for the whole cluster. Regardless of vhost chosen for current logged in user. 

- Add file vhost definition upload endpoint in API.
- No vhost detected in session storage, pick default for user and save to storage.
- Change vhost in upper right menu (`userMenuVhost`), update session storage with vhost.
- Export definition drop down, set to vhost in session storage.
- Import definition, set to vhost in session storage.
	- All, upload defintions to whole cluster (require administrator tag)
	- <vhost>, upload definition to vhost (require poliymaker/administrator tag)
	
![Screenshot 2022-11-15 at 08 35 21](https://user-images.githubusercontent.com/47317658/201862405-8b605f3e-766c-4666-908c-531566a7250a.png)

carlhoerberg commented:
I think the vhost selector should be next to the upload button, i think it can easliy be overlooked otherwise. Someone selected a vhost when viewing queues, than switches over to the overview page and upload defintions, thinkig that it is for the whole vhost and then is left wondering why it didn't happen.

tbroden84 commented:
True, would be a lot better.

tbroden84 commented:
Removed fetching vhost from session storage and added separate vhost selector for uploading definitions. Either upload defintions to cluster or only for selected vhost.

![Screenshot 2022-11-22 at 09 40 12](https://user-images.githubusercontent.com/47317658/203267495-c06993c7-c40b-4600-b123-5607041a0d53.png)

carlhoerberg commented:
very nice!

On Tue, Nov 22, 2022 at 9:47 AM Tobias Brodén ***@***.***>
wrote:

> Removed fetching vhost from session storage and added separate vhost
> selector for uploading definitions. Either upload defintions to cluster or
> only for selected vhost.
>
> [image: Screenshot 2022-11-22 at 09 40 12]
> <https://user-images.githubusercontent.com/47317658/203267495-c06993c7-c40b-4600-b123-5607041a0d53.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cloudamqp/lavinmq/pull/409#issuecomment-1323308215>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABL6TSXJK6RMST4MIBO6SLWJSCCNANCNFSM6AAAAAASAUSOUU>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
